desc/compute/_surface.py

```diff
@@ -427,6 +427,173 @@ def _e_zeta_z_FourierRZToroidalSurface(params, transforms, profiles, data, **kwa
+    basis="basis",
+)
+def _Phi_z_FourierCurrentPotentialField(params, transforms, profiles, data, **kwargs):
+    data["Phi_z"] = (
+        transforms["Phi"].transform(params["Phi_mn"], dz=1) + params["G"] / 2 / jnp.pi
+    )
+    return data
+
+
+@register_compute_fun(
+    name="Phi",
+    label="\\Phi",
+    units="A",
+    units_long="amperes",
+    description="Surface current potential",
+    dim=2,
+    params=["params"],
+    transforms={"grid": [], "potential": []},
+    profiles=[],
+    coordinates="tz",
+    data=[],
+    parameterization="desc.magnetic_fields.CurrentPotentialField",
+    basis="basis",
+)
+def _Phi_CurrentPotentialField(params, transforms, profiles, data, **kwargs):
+    data["Phi"] = transforms["potential"](
+        transforms["grid"].nodes[:, 1],
+        transforms["grid"].nodes[:, 2],
+        **params["params"][0]
+    )
+    return data
+
+
+@register_compute_fun(
+    name="Phi_t",
+    label="\\partial_{\\theta}Phi",
+    units="A",
+    units_long="amperes",
+    description="Surface current potential, poloidal derivative",
+    dim=2,
+    params=["params"],
+    transforms={"grid": [], "potential_dtheta": []},
+    profiles=[],
+    coordinates="tz",
+    data=[],
+    parameterization="desc.magnetic_fields.CurrentPotentialField",
+    basis="basis",
+)
+def _Phi_t_CurrentPotentialField(params, transforms, profiles, data, **kwargs):
+    data["Phi_t"] = transforms["potential_dtheta"](
+        transforms["grid"].nodes[:, 1],
+        transforms["grid"].nodes[:, 2],
+        **params["params"][0]
+    )
+    return data
+
+
+@register_compute_fun(
+    name="Phi_z",
+    label="\\partial_{\\zeta}Phi",
+    units="A",
+    units_long="amperes",
+    description="Surface current potential, toroidal derivative",
+    dim=2,
+    params=["params"],
+    transforms={"grid": [], "potential_dzeta": []},
+    profiles=[],
+    coordinates="tz",
+    data=[],
+    parameterization="desc.magnetic_fields.CurrentPotentialField",
+    basis="basis",
+)
+def _Phi_z_CurrentPotentialField(params, transforms, profiles, data, **kwargs):
+    data["Phi_z"] = transforms["potential_dzeta"](
+        transforms["grid"].nodes[:, 1],
+        transforms["grid"].nodes[:, 2],
+        **params["params"][0]
+    )
+    return data
+
+
+@register_compute_fun(
+    name="K",
+    label="\\mathbf{K}",
+    units="A",
+    units_long="amperes",
+    description="Surface current density, defined as the"
+    " surface normal vector cross gradient of current potential",
+    dim=2,
+    params=[],
+    transforms={},
+    profiles=[],
+    coordinates="tz",
+    data=["Phi_t", "Phi_z", "x", "e_theta", "e_zeta", "|e_theta x e_zeta|"],
+    parameterization=[
+        "desc.magnetic_fields.CurrentPotentialField",
+        "desc.magnetic_fields.FourierCurrentPotentialField",
+    ],
+    basis="basis",
+)
+def _K_CurrentPotentialField(params, transforms, profiles, data, **kwargs):
+    data["K"] = (
```

Comment:
would it be clearer to have an intermediate qty for `grad(Phi)` and then do `cross(n, grad(Phi))`?

Reply:
Maybe, but the contravariant basis vectors are a bit hard to define on the surface, and afaik it involves inverting the covariant metric tensor. It is easiest to go straight to the `n x grad(phi)` expression as then all the contravariant vectors become covariant basis vectors and those are immediately defined for a surface.